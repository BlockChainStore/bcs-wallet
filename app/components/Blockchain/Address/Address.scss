@import '../../../styles/variables';

.address {
    color: $link-color;
    cursor: pointer;
}
