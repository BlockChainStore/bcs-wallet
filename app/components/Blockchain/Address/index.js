// @flow
import { compose } from 'recompose'

import Address from './Address'
import withNetworkData from '../../../hocs/withNetworkData'
import withExplorerData from '../../../hocs/withExplorerData'

export default compose(
  withNetworkData(),
  withExplorerData()
)(Address)
