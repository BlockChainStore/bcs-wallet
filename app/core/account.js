// @flow
import { wallet } from 'neon-js'

import { getStorage, setStorage } from './storage'

function getNEP6AddressData (matchesEncryptedWIF: boolean, address: string) {
  if (matchesEncryptedWIF) {
    return { address }
  } else {
    return {}
  }
}

export async function upgradeNEP6AddAddresses (encryptedWIF: string, wif: string) {
  const data = getStorage('userWallet')
  const loggedIntoAccount = new wallet.Account(wif)

  if (data && data.accounts) {
    const accounts = data.accounts.map((account, idx) => ({
      ...account,
      ...getNEP6AddressData(account.key === encryptedWIF, loggedIntoAccount.address)
    }))

    await setStorage('userWallet', { ...data, accounts })
  }
}
