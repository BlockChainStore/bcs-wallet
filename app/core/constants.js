// @flow
export const NEON_WALLET_RELEASE_LINK =
  'https://github.com/CityOfZion/neon-wallet/releases'

export const ASSETS = {
  NEO: 'NEO',
  GAS: 'GAS'
}

export const EXPLORERS = {
  NEO_TRACKER: 'Neotracker',
  NEO_SCAN: 'Neoscan',
  ANT_CHAIN: 'Antchain',
  NEO_VERSE: 'NEOverse'
}

export const ROUTES = {
  HOME: '',
  DASHBOARD: '/dashboard',
  CREATE_WALLET: '/create',
  ENCRYPT_KEY: '/encrypt-key',
  LOGIN_PRIVATE_KEY: '/login-private-key',
  LOGIN_LOCAL_STORAGE: '/login-local-storage',
  LOGIN_LEDGER_NANO_S: '/login-ledger-nano-s',
  LOGIN_NEP2: '/login-encrypted',
  LOGIN_TOKEN_SALE: '/login-token-sale',
  TOKEN_SALE: '/token-sale',
  SETTINGS: '/settings',
  DISPLAY_WALLET_KEYS: '/display-wallet-keys'
}

export const NOTIFICATION_LEVELS = {
  ERROR: 'error',
  SUCCESS: 'success',
  INFO: 'info',
  WARNING: 'warning'
}

export const NOTIFICATION_POSITIONS = {
  TOP_CENTER: 'tc',
  TOP_RIGHT: 'tr',
  TOP_LEFT: 'tl',
  BOTTOM_CENTER: 'bc',
  BOTTOM_RIGHT: 'br',
  BOTTOM_LEFT: 'bl'
}

export const BIP44_PATH =
  '8000002C' + '80000378' + '80000000' + '00000000' + '00000000'

export const MODAL_TYPES = {
  SEND: 'SEND',
  RECEIVE: 'RECEIVE',
  CONFIRM: 'CONFIRM',
  TOKEN_INFO: 'TOKEN_INFO',
  TOKEN: 'TOKEN',
  ICO: 'ICO'
}

export const MAIN_NETWORK_ID = '1'
export const TEST_NETWORK_ID = '2'

// TestNet
export const TOKENS_TEST = {
  BCS : 'e81c837d37763afbf894d319c868a09f219f2be9'
}

// MainNet
export const TOKENS = {
  BCS : '546c5872a992b2754ef327154f4c119baabff65f'
}

export const DEFAULT_WALLET = {
  name: 'userWallet',
  version: '1.0',
  scrypt: {
    cost: 16384,
    blockSize: 8,
    parallel: 8,
    size: 64
  },
  accounts: [],
  extra: null
}

export const DEFAULT_CURRENCY_CODE = 'usd'

export const CURRENCIES = {
  aud: { symbol: '$' },
  brl: { symbol: 'R$' },
  cad: { symbol: '$' },
  chf: { symbol: 'Fr.' },
  clp: { symbol: '$' },
  cny: { symbol: '¥' },
  czk: { symbol: 'Kč' },
  dkk: { symbol: 'kr. ' },
  eur: { symbol: '€' },
  gbp: { symbol: '£' },
  hkd: { symbol: '$' },
  huf: { symbol: 'Ft ' },
  idr: { symbol: 'Rp ' },
  ils: { symbol: '₪' },
  inr: { symbol: '₹' },
  jpy: { symbol: '¥' },
  krw: { symbol: '₩' },
  mxn: { symbol: '$' },
  myr: { symbol: 'RM' },
  nok: { symbol: 'kr ' },
  nzd: { symbol: '$' },
  php: { symbol: '₱' },
  pkr: { symbol: '₨ ' },
  pln: { symbol: 'zł' },
  rub: { symbol: '₽' },
  sek: { symbol: 'kr ' },
  sgd: { symbol: 'S$' },
  thb: { symbol: '฿' },
  try: { symbol: '₺' },
  twd: { symbol: 'NT$' },
  usd: { symbol: '$' },
  zar: { symbol: 'R ' }
}
