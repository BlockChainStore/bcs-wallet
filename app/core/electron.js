// @flow
import { shell } from 'electron'

export const openExternal = (srcLink: string) => {
  shell.openExternal(srcLink)
}
