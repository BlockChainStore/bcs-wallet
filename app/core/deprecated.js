// @flow
import { get, omit, pick } from 'lodash'

import { ID as AUTH_ID } from '../actions/authActions'
import { ID as BALANCES_ID } from '../actions/balancesActions'
import { ID as NETWORK_ID } from '../actions/networkActions'
import { ASSETS } from '../core/constants'
import { findNetwork } from '../core/networks'

const PREFIX = 'api'

// TODO: Module functions that inspect state such as `doClaimNotify` should be refactored into
//       higher-order components that expose function props for performing these actions.  Data that
//       is normally provided by functions like `getNetwork` can instead be retrieved via the
//       `withData` higher-order component.

export const getNetworkId = (state: Object) => {
  return get(state, `${PREFIX}.${NETWORK_ID}.data`)
}

export const getNetwork = (state: Object) => {
  return getNetworkById(getNetworkId(state))
}

export const getNetworkById = (networkId: string) => {
  return findNetwork(networkId).network
}

export const getAddress = (state: Object) => {
  return get(state, `${PREFIX}.${AUTH_ID}.data.address`)
}

export const getWIF = (state: Object) => {
  return get(state, `${PREFIX}.${AUTH_ID}.data.wif`)
}

export const getSigningFunction = (state: Object) => {
  return get(state, `${PREFIX}.${AUTH_ID}.data.signingFunction`)
}

export const getPublicKey = (state: Object) => {
  return get(state, `${PREFIX}.${AUTH_ID}.data.publicKey`)
}

export const getIsHardwareLogin = (state: Object) => {
  return get(state, `${PREFIX}.${AUTH_ID}.data.isHardwareLogin`)
}

export const getNEO = (state: Object): string => {
  return getBalances(state)[ASSETS.NEO]
}

export const getGAS = (state: Object): string => {
  return getBalances(state)[ASSETS.GAS]
}

export const getTokenBalances = (state: Object): Object => {
  return omit(getBalances(state), ASSETS.NEO, ASSETS.GAS)
}

export const getAssetBalances = (state: Object): Object => {
  return pick(getBalances(state), ASSETS.NEO, ASSETS.GAS)
}

export const getBalances = (state: Object) => {
  return get(state, `${PREFIX}.${BALANCES_ID}.data`)
}
