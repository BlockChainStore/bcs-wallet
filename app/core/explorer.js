// @flow
import { openExternal } from './electron'
import { EXPLORERS } from '../core/constants'
import { isMainNetwork } from '../core/networks'

export const getExplorerBaseURL = (networkId: string, explorer: ExplorerType) => {
  let baseURL
  if (explorer === EXPLORERS.NEO_TRACKER) {
    if (isMainNetwork(networkId)) {
      baseURL = 'https://neotracker.io'
    } else {
      baseURL = 'https://testnet.neotracker.io'
    }
  } else if (explorer === EXPLORERS.NEO_SCAN) {
    if (isMainNetwork(networkId)) {
      baseURL = 'https://neoscan.io'
    } else {
      baseURL = 'https://neoscan-testnet.io'
    }
  } else if (explorer === EXPLORERS.NEO_VERSE) {
    if (isMainNetwork(networkId)) {
      baseURL = 'http://explorer.neoverse.io'
    } else {
      baseURL = 'http://testnet.neoverse.io'
    }
  } else {
    if (isMainNetwork(networkId)) {
      baseURL = 'http://antcha.in'
    } else {
      baseURL = 'http://testnet.antcha.in'
    }
  }
  return baseURL
}

export const getExplorerTxLink = (networkId: string, explorer: ExplorerType, txId: string) => {
  const baseURL = getExplorerBaseURL(networkId, explorer)

  if (explorer === EXPLORERS.NEO_TRACKER) {
    return `${baseURL}/tx/${txId}`
  } else if (explorer === EXPLORERS.NEO_SCAN) {
    return `${baseURL}/transaction/${txId}`
  } else if (explorer === EXPLORERS.NEO_VERSE) {
    return `${baseURL}/transactions/${txId}`
  } else {
    return `${baseURL}/tx/hash/${txId}`
  }
}

export const getExplorerAddressLink = (networkId: string, explorer: ExplorerType, address: string) => {
  const baseURL = getExplorerBaseURL(networkId, explorer)

  if (explorer === EXPLORERS.NEO_TRACKER) {
    return `${baseURL}/address/${address}`
  } else if (explorer === EXPLORERS.NEO_SCAN) {
    return `${baseURL}/address/${address}/1`
  } else if (explorer === EXPLORERS.NEO_VERSE) {
    return `${baseURL}/addresses/${address}`
  } else {
    return `${baseURL}/address/info/${address}`
  }
}

export const getExplorerAssetLink = (networkId: string, explorer: ExplorerType, assetId: string) => {
  const baseURL = getExplorerBaseURL(networkId, explorer)

  if (explorer === EXPLORERS.NEO_TRACKER) {
    return `${baseURL}/asset/${assetId}`
  } else if (explorer === EXPLORERS.NEO_SCAN) {
    return `${baseURL}/asset/${assetId}`
  } else if (explorer === EXPLORERS.NEO_VERSE) {
    return `${baseURL}/assets/${assetId}`
  } else {
    return `${baseURL}/asset/hash/${assetId}`
  }
}

export const openExplorerTx = (networkId: string, explorer: ExplorerType, txId: string) =>
  openExternal(getExplorerTxLink(networkId, explorer, txId))

export const openExplorerAddress = (networkId: string, explorer: ExplorerType, address: string) =>
  openExternal(getExplorerAddressLink(networkId, explorer, address))

export const openExplorerAsset = (networkId: string, explorer: ExplorerType, assetId: string) =>
  openExternal(getExplorerAssetLink(networkId, explorer, assetId))
