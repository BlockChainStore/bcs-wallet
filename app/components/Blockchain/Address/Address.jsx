// @flow
import React from 'react'
import classNames from 'classnames'

import { openExplorerAddress } from '../../../core/explorer'
import styles from './Address.scss'

type Props = {
  className?: string,
  networkId: NetworkType,
  explorer: ExplorerType,
  address: string
}

export default class Address extends React.Component<Props> {
  render = () => {
    const { address, className } = this.props

    return (
      <span className={classNames(styles.address, className)} onClick={this.handleClick}>
        {address}
      </span>
    )
  }

  handleClick = () => {
    const { networkId, explorer, address } = this.props
    openExplorerAddress(networkId, explorer, address)
  }
}
