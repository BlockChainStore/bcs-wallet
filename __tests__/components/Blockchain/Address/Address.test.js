import React from 'react'
import { shallow } from 'enzyme'

import Address from '../../../../app/components/Blockchain/Address/Address'
import { TEST_NETWORK_ID, EXPLORERS } from '../../../../app/core/constants'

describe('Address', () => {
  const props = {
    address: 'AW4FD7bz6PF2QadFKF8qXUT7tNmWgvXZc4',
    networkId: TEST_NETWORK_ID,
    explorer: EXPLORERS.NEO_SCAN
  }

  test('should render without crashing', () => {
    const wrapper = shallow(<Address {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
