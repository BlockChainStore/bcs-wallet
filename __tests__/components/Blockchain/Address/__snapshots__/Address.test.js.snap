// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address should render without crashing 1`] = `
<span
  className="address"
  onClick={[Function]}
>
  AW4FD7bz6PF2QadFKF8qXUT7tNmWgvXZc4
</span>
`;
